src/hooks/useWatchlist.js

```diff
@@ -1,37 +1,68 @@
-import { useState, useEffect } from 'react';
-import { saveToLocal, loadFromLocal } from '../utils/localStorage';
+import useSWR from 'swr';
 
-export default function useWatchlist() {
-  const [watchlist, setWatchlist] = useState(loadFromLocal('watchlist') ?? []);
+const fetcher = (...args) => fetch(...args).then(res => res.json());
 
-  useEffect(() => {
-    saveToLocal('watchlist', watchlist);
-  }, [watchlist]);
+export default function useWatchlist() {
+  const {
+    data: watchlist,
+    error: watchlistError,
+    mutate: mutateWatchlist,
+  } = useSWR('/api/watchlist', fetcher);
 
-  function handleAddMovie(id, title, posterPath) {
+  async function handleAddMovie(id, title, posterPath) {
     const watchlistItem = { id, title, posterPath };
     if (watchlist.find(item => item.id === watchlistItem.id)) {
-      setWatchlist([...watchlist]);
+      mutateWatchlist([...watchlist]);
     } else {
-      setWatchlist([watchlistItem, ...watchlist]);
+      mutateWatchlist([...watchlist, watchlistItem], false);
+      await fetch('/api/watchlist', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(watchlistItem),
+      });
+      mutateWatchlist();
     }
   }
 
-  function handleAddSeries(id, name, posterPath) {
+  async function handleAddSeries(id, name, posterPath) {
     const watchlistItem = { id, name, posterPath };
-    if (watchlist.find(item => item.id === watchlistItem.id)) {
-      setWatchlist([...watchlist]);
+    if (watchlist?.find(item => item.id === watchlistItem.id)) {
+      mutateWatchlist([...watchlist]);
     } else {
-      setWatchlist([watchlistItem, ...watchlist]);
+      mutateWatchlist([...watchlist, watchlistItem], false);
+      await fetch('/api/watchlist', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(watchlistItem),
+      });
+      mutateWatchlist();
     }
   }
 
-  function handleDeleteItem(id) {
-    setWatchlist(watchlist.filter(item => item.id !== id));
+  async function handleDeleteItem(id) {
+    const filteredItems = watchlist?.filter(result => result.id !== id);
+
+    mutateWatchlist(filteredItems, false);
+
+    const filteredItem = watchlist?.filter(result => result.id === id);
+    const deleteId = filteredItem[0]._id;
+
+    await fetch('/api/watchlist', {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ deleteId }),
+    });
+    mutateWatchlist();
   }
```

Comment:
The `id` you get as a parameter, is this not the same as `deleteId` on line 59? It seems a bit unnecessary to do all the filtering 😅 😬

Reply:
I unfortunately need both id,
I need the `_id` from the backend to be able to delete the entry.
And with `id` I only get to the detail page for a movie or series.